### DISCUSSION TITLE
How to get my Render(...) graphics to go over Background

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#9867] [General]
I've got 

    public class GraphGrid : ContentPresenter
    {
        public override void Render(DrawingContext dc)
        {
            base.Render(dc);
            // draw stuff
        }
    }

Currently all the custom graphics appear under the Background.  Changing whether `base.Render(dc)` goes first or last doesn't seem to make a difference.  How do I get Avalonia to draw the Background first, so my custom graphics appear on top?

Or alternately, how do I get Avalonia to skip drawing the Background so I can draw it in my `Render(...)` method?  And how would I actually make that call, given that Background could be any number of things?

thanks

**Comment by timunie** [comment]
Hi @sonicsmooth ,

first of all I wonder why you use `ContentPresenter` as your base class. Normally I'd use `Control` for my own rendering or `ContentControl` if your need to add any content to it. `ContentPresenter` is a special one mainly for usage in `ControlTemplates`. 

You may also want to check out the Avalonia source and have a look into the Render-Demos in order to see how it can be done.

**Reply by sonicsmooth** [reply]
Thanks, I made an error in my post.  The class is derived from `ContentControl` indeed.  I've been playing around with Control, ContentControl, and a few others.  For this one I am using a control template and wish to be able to specify background, border, etc., at the time of instantiaton.  Because I'm catching pointer events for pan/zoom, I need a background to be present, so I defined a transparent one here as default, but it can be overridden at instantiation and/or by a separate styles file.  I will eventually have content in this window, or perhaps layer this grid behind the actual content.  My template looks like the following.  Also I couldn't get the `<Design.PreviewWidth.../>` to preview in Visual Studio; it will only display built-in controls in the previewer.  I'm guessing I need to specify this axaml file someplace for that to work.  Currently this file is listed in App.axaml.

    <Styles xmlns="https://github.com/avaloniaui"
            xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
            xmlns:controls="using:avgrid.Controls">

      <Design.PreviewWith>
        <controls:GraphGrid>
          <Button>Click Graph</Button>
        </controls:GraphGrid>
      </Design.PreviewWith>

      <Style Selector="controls|GraphGrid">
        
        <Setter Property="Background" Value="Transparent"/>
        <Setter Property="Template">
          <ControlTemplate>
            <ContentPresenter Content="{TemplateBinding Content}"
                              Background="{TemplateBinding Background}"
                              BorderBrush="{TemplateBinding BorderBrush}"
                              BorderThickness="{TemplateBinding BorderThickness}"/>
          </ControlTemplate>
        </Setter>
      </Style>

**Reply by timunie** [reply]
just wrap your ContentPresneter in a Border or Panel woith Transparent Background
